packages/frontend-api/src/Model/Parameter/ParameterWithValuesFactory.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Parameter;
+
+use Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Product\ProductCachedAttributesFacade;
+
+class ParameterWithValuesFactory
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\ProductCachedAttributesFacade
+     */
+    protected $productCachedAttributesFacade;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductCachedAttributesFacade $productCachedAttributesFacade
+     */
+    public function __construct(
+        ProductCachedAttributesFacade $productCachedAttributesFacade
+    ) {
+        $this->productCachedAttributesFacade = $productCachedAttributesFacade;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter $parameter
+     * @param \Shopsys\FrameworkBundle\Model\Product\Parameter\ProductParameterValue[] $parameterValues
+     * @return \Shopsys\FrontendApiBundle\Model\Parameter\ParameterWithValues
+     */
+    public function create(Parameter $parameter, array $parameterValues): ParameterWithValues
+    {
+        return new ParameterWithValues($parameter, $parameterValues);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @return \Shopsys\FrontendApiBundle\Model\Parameter\ParameterWithValues[]
+     */
+    public function createMultipleForProduct(Product $product): array
+    {
+        $productParameterValues = $this->productCachedAttributesFacade->getProductParameterValues($product);
+
+        $valuesByParameterId = [];
+        $parameters = [];
+        $parametersWithValues = [];
+
+        foreach ($productParameterValues as $productParameterValue) {
+            $valuesByParameterId[$productParameterValue->getParameter()->getId()][] = $productParameterValue->getValue();
+            $parameters[$productParameterValue->getParameter()->getId()] = $productParameterValue->getParameter();
```

Comment:
Just a nitpick to make it more readable.

```suggestion
            $parameterId = $productParameterValue->getParameter()->getId();
            $valuesByParameterId[parameterId][] = $productParameterValue->getValue();
            $parameters[parameterId] = $productParameterValue->getParameter();
```

Reply:
done